internal/queue/queued_channel.go

```diff
@@ -30,14 +41,52 @@ func NewQueuedChannel[T any](channelBufferSize int, capacity int) *QueuedChannel
 	return queue
 }
 
-func (q *QueuedChannel[T]) Queue(items ...T) bool {
-	return q.queue.PushMany(items...)
+func (q *QueuedChannel[T]) Enqueue(items ...T) bool {
+	if q.closed.Load().(bool) {
```

Comment:
Can we switch this back to the old code which uses the uint32 atomic functions so we don't have to do type casts?

Reply:
I took the atomic.Bool implementation from Go 1.19 stdlib, i think that's the best option for now, then we switch to official atomic.Bool later